spyder/widgets/editor.py

```diff
@@ -304,7 +304,12 @@ def __len__(self):
         return len(self.history)
 
     def __getitem__(self, i):
-        return self.id_list.index(self.history[i])
+        self._update_id_list()
+        try:
+            return self.id_list.index(self.history[i])
+        except ValueError:
+            self.refresh
+            raise IndexError

```

Comment:
Do we really need to raise an `IndexError` here?

Reply:
I think it's the expected bevahior, when an elementary isn't found in a list an IndexError is raised.
The other option will be to return None.